src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -21,46 +27,100 @@ public DiagController(ILogger<DiagController> logger, IDiagnosticServices diagno
             _diagnosticServices = diagnosticServices;
         }
 
-        [HttpGet("getpids")]
-        public ActionResult<IEnumerable<string>> GetPids()
+        [HttpGet("processes")]
+        public ActionResult<IEnumerable<ProcessModel>> GetProcesses()
         {
-            return new JsonResult(_diagnosticServices.GetProcesses());
+            return InvokeService(() =>
+            {
+                IList<ProcessModel> processes = new List<ProcessModel>();
+                foreach (int pid in _diagnosticServices.GetProcesses())
+                {
+                    processes.Add(new ProcessModel() { Pid = pid });
+                }
+                return new ActionResult<IEnumerable<ProcessModel>>(processes);
+            });
         }
 
-        [HttpGet(@"{pid}/dump")]
-        public Task<ActionResult> GetDump(int pid, [FromQuery]int? dumpType = 1)
+        [HttpGet("dump")]
+        public Task<ActionResult> GetDump([FromQuery] DumpType type = DumpType.WithHeap)
         {
-            if (!Enum.IsDefined(typeof(DumpType), dumpType))
-            {
-                return Task.FromResult<ActionResult>(BadRequest(new ProblemDetails { Detail = "Invalid dump type", Status = 400 }));
-            }
+            return GetDumpImpl(pid: null, type);
+        }
 
+        [HttpGet("dump/{pid}")]
+        public Task<ActionResult> GetDump(int pid, [FromQuery]DumpType type = DumpType.WithHeap)
+        {
+            return GetDumpImpl(pid, type);
+        }
+
+        public Task<ActionResult> GetDumpImpl(int? pid, DumpType type)
+        {
             return InvokeService(async () =>
             {
-                Stream dump = await _diagnosticServices.GetDump(pid, (DumpType)dumpType);
+                OperationResult<Stream> result = await _diagnosticServices.GetDump(pid, type);
 
                 //Compression is done automatically by the response
                 //Chunking is done because the result has no content-length
-                return File(dump, "application/octet-stream", fileDownloadName: FormattableString.Invariant($"coredump_{pid}"));
+                return CreateFileStreamResult(result, $"coredump_{result.Pid}{DumpFileExtension}");
```

Comment:
>coredump_ [](start = 56, length = 9)

maybe a windowns specific name as well?

Reply:
On Windows, the WER service records dump names as <processname>.<processext>.<processid>.dmp and Visual Studio captures dumps as <processname>.dmp. I can use Process.GetProcessById to get the process information, but this will only work if the process is in the same process space as dotnet-monitor (which is the case at this time). When we enable the use of the diagnostic api server, then we likely won't have access to this information. So we'll need a fallback either way. But I can add in this functionality at this time just merely adopt the fallback for the diagnostic api server scenario when that comes online.